authors/apps/articles/views.py

```diff
@@ -43,3 +50,46 @@ def destroy(self, request, *args, **kwargs):
         self.perform_destroy(instance)
         return Response({"message": "category deleted"},
                         status=status.HTTP_204_NO_CONTENT)
+
+
+class CreateArticle(ListCreateAPIView):
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ArticleJSONRenderer,)
+    serializer_class = ArticleSerializer
+    queryset = Article.objects.all()
+
+    def create(self, request):
+
+        article = request.data.get('article', {})
+        current_user = User.objects.get(email=request.user)
```

Comment:
@Stanley-Okwii I think this can be best done by overriding the perform_create method with **serializer.save(author=self.request.user__profile)** than doing a query back to the database.

Reply:
@PatrickCmd , I have worked on this. Thanks for the feedback